### PR TITLE
Fix No Contact (14 Days) Count

### LINKED ISSUE #850: Contacts made per volunteer per supervisor not reflecting in Admin's supervisor view
As an Admin,
Within the Supervisor view,
I want to see an accurate count of volunteers per supervisor (with active case assignments) who have not made contact with their `casa_case` (including non `youth` `contact_types`) in 14 days or more, 
so that I can stay abreast of volunteer activity.

**How it works now**
An incorrect count of volunteers who have not made contact with their `casa_case` currently displays in the "No Contact (14 days)" column of the Supervisor view, accessible by admins only.

**How it should work**
An accurate count of volunteers who **have not made contact** with at least one of any assigned `casa_cases` should display in the column "No Contact (14 days)" in the Supervisor view. 

**Which volunteers should be accounted for in this column?**
Volunteers who are `active` and
have at least one `active` case assignment
and are assigned to a supervisor
and have not created a `case_contact` in 14 days or more
or have only created `case_contacts` in which `contact_made = false` for 14 days or more.

**How to test**
- <a href="https://casa-r4g-staging.herokuapp.com/">Sign in to staging</a> as casa_admin1@example.com (pw: 123456). 
- Scroll down to "Supervisors".
- Make note of how many volunteers assigned to S u p e r v i s o r ONE are in the column: "No Contact (14 days)"
- Scroll up to "Volunteers". 
- Filter by: Supervisor = S u p e r v i s o r ONE
- Identify a volunteer assigned to S u p e r v i s o r ONE that is only assigned to one Case Number and has a clock icon next to their name. 
- Click on the volunteer's name.
- Copy their email.
- Log out.
- Log back in using the volunteer's email you copied, and the password: 123456
- Create a new case contact. Mark "contact made" as "yes". 
- Log out. 
- Log back in as casa_admin1@example.com using the pw: 123456
- Scroll back down to "Supervisors".
- The number under "No Contact (14 days)" in the row belonging to S u p e r v i s o r ONE remains the same, even though it should have decreased by one. 

**Acceptance Criteria**
After following the above instructions in _How to test_, the number in the final step increases by one.

### PR DESCRIPTION
**Pull request by ryanmrodriguez**
### What github issue is this PR for, if any?
Resolves #850 

### What changed, and why?
*Fixed* an incorrect count of volunteers who have not made contact with their casa_case currently displays in the "No Contact (14 days)" column of the Supervisor view, accessible by admins only.

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪
Added 5 rspec test cases

### Screenshots please :)
No visual changes

### Feelings gif (optional)
Womp
Eggs

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -79,9 +79,19 @@ def volunteers_serving_transistion_aged_youth
   end
 
   def no_contact_for_two_weeks
-    volunteers.includes(:case_contacts)
-      .where(case_contacts: {contact_made: true})
-      .where.not(case_contacts: {occurred_at: 14.days.ago..Date.today}).size
+    # Get ACTIVE volunteers that have ACTIVE supervisor assignments with at least one ACTIVE case
```
Comment (compwron) [inline app/models/user.rb:82]: Excellent comment :)

**Review by compwron — APPROVED** [review]
YAY TESTS
That is an impressive query 😂🙌🏼✨

**Review by jmkoni — APPROVED** [review]
Great work on this beast of a query!

**Comment by KatherineMuedas** [comment]
Kudos to @ryanmrodriguez for the query 🚀🚀🚀🚀  He is really a master of queries! Really impressive!! ⭐️👌🏻

**Pull request merged**